utilities/pytorch_model.py

```diff
@@ -40,31 +45,67 @@ def __init__(self,
         self.seed = seed
         torch.manual_seed(self.seed)
         np.random.seed(self.seed)
-        if torch.backends.cudnn.enabled:
+        if use_gpu and torch.backends.cudnn.enabled:
             torch.backends.cudnn.deterministic = True
             torch.backends.cudnn.benchmark = False
 
+        # if both network_file and network_penalty are provided, load network
+        # data and calculate graph Laplacian
+        if network_file is not None and params_map['network_penalty'] != 0:
+            assert network_features is not None, (
+                'list of features in network should be included')
+            self.network_features = network_features
+            import networkx as nx
+            G = nx.read_weighted_edgelist(network_file, delimiter='\t')
+            lt = nx.laplacian_matrix(G)
+            indices, values, shape = self._convert_csr_to_sparse_inputs(lt)
```

Comment:
I'll add a note to the code, but in case you're curious: it's a format for storing sparse matrices in a memory-efficient way ([see here](https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_row_\(CSR,_CRS_or_Yale_format\)) for more details). PyTorch uses a different format, so I have to convert it here.